src/app/clusters/general-diagnostics-server/general-diagnostics-server.cpp

```diff
@@ -192,7 +197,9 @@ CHIP_ERROR GeneralDiagosticsAttrAccess::Read(const ConcreteReadAttributePath & a
         return ReadIfSupported(&DiagnosticDataProvider::GetRebootCount, aEncoder);
     }
     case UpTime::Id: {
-        return ReadIfSupported(&DiagnosticDataProvider::GetUpTime, aEncoder);
+        System::Clock::Seconds64 system_time_seconds =
```

Comment:
This and other places where we duration_cast _might_ be a good use of `auto`.   Hard to say for sure.

Reply:
Prefer to use the correct unit explicitly rather than auto. If really important will do as a follow-up.

Reply:
Like I said: hard to say for sure.  On the one hand, it repeats the type that's right there in the cast.  On the other, it ensures we have the right type.  I'm fine either way here.